### PR TITLE
DDF-2131: removed anytags query from catalogframework

### PR DESCRIPTION
**Pull request by rzwiefel**
#### What does this PR do?
Fixes compatibility issues by removing the anytag query from inside catalogframework and instead uses a flag in caching federation strategy
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@harrison-tarr 
@ryeats 
@tbatie 
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@coyotesqrl
@pklinef
@stustison

### CODE REVIEW

File: catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/CachingFederationStrategy.java [hunk 1]
```diff
@@ -367,6 +368,13 @@ private QueryRequest getRequestWithTagsFilter(QueryRequest request) {
         return newRequest;
     }
 
+    private boolean noDefaultTags(Operation op) {
+        return Optional.ofNullable(op)
+                    .map(Operation::getProperties)
+                    .map(p -> (Boolean) p.getOrDefault("no-default-tags", false))
+                    .orElse(false);
```
Comment (rzwiefel) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/CachingFederationStrategy.java]: @pklinef the `"no-default-tags"` should probably be a constant somewhere if this were a for sure fix, but if this is just a temporary thats probably overkill

Comment (coyotesqrl) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/CachingFederationStrategy.java]: Couldn't you skip the `getOrDefault` on the map as you've an `orElse` at the end anyway?

Comment (rzwiefel) [inline catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/CachingFederationStrategy.java]: yes i could
done

**Comment by shaundmorris** [comment]
LGTM based on what I have seen in the 2.9.x branch

**Pull request merged**